extensions/wikia/Email/EmailController.class.php

```diff
@@ -507,15 +507,13 @@ protected function getUserFromName( $username ) {
 			throw new Fatal( 'Required username has been left empty' );
 		}
 
-		// Allow an anonymous user to be specified
-		if ( $username == self::ANONYMOUS_USER_ID ) {
-			return \User::newFromId( 0 );
-		}
-
 		if ( $username instanceof \User ) {
 			$user = $username;
 		} else if ( is_object( $username ) ) {
 			throw new Fatal( 'Non-user object passed when user object or username expected' );
+		} else if ( $username == self::ANONYMOUS_USER_ID ) {
+			// Allow an anonymous user to be specified
+			return \User::newFromId( 0 );
 		} else {
 			$user = \User::newFromName( $username, $validate = false );
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this unused "$validate" local variable. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://10.8.74.32:9000/coding_rules#rule_key=php%3AS1481)